internal/cleanup/cleanup_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package cleanup
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/iam"
+	"github.com/hashicorp/boundary/internal/scheduler"
+	"github.com/hashicorp/boundary/internal/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrototypeInsertDeleteAndClear(t *testing.T) {
+	ctx := context.Background()
+
+	conn, _ := db.TestSetup(t, "postgres")
+	wrapper := db.TestWrapper(t)
+	iamRepo := iam.TestRepo(t, conn, wrapper)
+	session := session.TestDefaultSession(t, conn, wrapper, iamRepo)
+	rw := db.New(conn)
+
+	db, err := conn.SqlDB(ctx)
+	if err != nil {
+		t.Errorf("error getting db connection %s", err)
+	}
+
+	_, err = db.Exec(fmt.Sprintf(`delete from target where public_id = '%s'`, session.TargetId))
+	if err != nil {
+		t.Errorf("error deleting from target %s", err)
+	}
+
+	// make sure that the trigger works
+	rows, err := db.Query("select count(public_id) from target_deleted")
+	if err != nil {
+		t.Errorf("error checking target_deleted table %s", err)
+	}
+	defer rows.Close()
+	var count int
+	for rows.Next() {
+		err = rw.ScanRows(ctx, rows, &count)
+		if err != nil {
+			t.Error("unable to scan rows for census sessions pending count")
+		}
+	}
+	require.Equal(t, 1, count)
+
+	sJob := cleanupJob{
+		r: rw,
+		w: rw,
+	}
+
+	err = sJob.Run(ctx)
+	require.NoError(t, err)
+
+	// make sure that the job doesn't clean up
+	rows, err = db.Query("select count(public_id) from target_deleted")
+	if err != nil {
+		t.Errorf("error checking target_deleted table %s", err)
+	}
+	defer rows.Close()
+	count = 0
+	for rows.Next() {
+		err = rw.ScanRows(ctx, rows, &count)
+		if err != nil {
+			t.Error("unable to scan rows for census sessions pending count")
+		}
+	}
+	require.Equal(t, 1, count)
+
+	_, err = db.Exec(`update target_deleted set delete_time = '2023-05-18';`)
```

Comment:
This will work, but we could just set it to `time.Now().AddDate(0, -2, 0)` to make it a little less weird.
```suggestion
	_, err = db.Exec("update target_deleted set delete_time = $1", time.Now().AddDate(0, -2, 0))
```